backend/core/logs/_posthog_processor.py

```diff
@@ -0,0 +1,93 @@
+import contextlib
+import os
+from typing import Any
+
+import posthog
+from structlog.types import EventDict, WrappedLogger
+
+
+class PostHogProcessor:
+    """PostHog processor for structlog.
+
+    Sends analytics events to PostHog when 'analytics' key is present in context.
+    """
+
+    def __init__(
+        self,
+        active: bool = True,
```

Comment:
remove active and verbose.
Instead, make the API key a part of the __init__ and only include the processor if the API key is defined.
You should also not setup the client globally but use `posthog = Posthog('<ph_project_api_key>', host='https://us.i.posthog.com')` as explained in the posthog doc